includes/Core/Authentication/Authentication.php

```diff
@@ -791,10 +781,39 @@ private function authentication_admin_notices( $notices ) {
 
 		$notices[] = $this->get_reauthentication_needed_notice();
 		$notices[] = $this->get_authentication_oauth_error_notice();
+		$notices[] = $this->get_reconnect_after_url_mismatch_notice();
 
 		return $notices;
 	}
 
+	/**
+	 * Gets reconnect notice.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Notice Notice object.
+	 */
+	private function get_reconnect_after_url_mismatch_notice() {
+		return new Notice(
+			'reconnect_after_url_mismatch',
+			array(
+				'content'         => function() {
+					return sprintf(
+						'<p>%s <a href="%s">%s</a></p>',
+						esc_html__( 'Looks like the URL of your site has changed. In order to continue using Site Kit, you\'ll need to reconnect, so that your plugin settings are updated with the new URL.', 'google-site-kit' ),
```

Comment:
```suggestion
						esc_html__( 'Looks like the URL of your site has changed. In order to continue using Site Kit, you’ll need to reconnect, so that your plugin settings are updated with the new URL.', 'google-site-kit' ),
```

Reply:
Updated.